recipes/mosquitto/2.x/conanfile.py

```diff
@@ -1,50 +1,71 @@
 import os
 
 from conans import ConanFile, CMake, tools
-from conans.errors import ConanInvalidConfiguration
+
+
+required_conan_version = ">=1.31.0"
+
 
 class Mosquitto(ConanFile):
     name = "mosquitto"
-    license = "EPL-2.0", "EPL-1.0"
+    license = "EPL-2.0"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://mosquitto.org"
     description = """Eclipse Mosquitto MQTT library, broker and more"""
     topics = ("MQTT", "IoT", "eclipse")
     settings = "os", "arch", "compiler", "build_type"
     options = {"shared": [True, False],
-                "with_tls": [True, False],
-                "clients": [True, False],
-                "broker": [True, False],
-                "apps": [True, False],
-                "plugins": [True, False],
-#                 "with_cjson": [True, False], TODO , adding a dependency is not enough, it needs also be found
+               "ssl": [True, False],
+               "clients": [True, False],
+               "broker": [True, False],
+               "apps": [True, False],
+               "cjson": [True, False],
+               "build_cpp": [True, False],
+               "websockets": [True, False],
             }
     default_options = {"shared": False,
-                        "with_tls": True,
-                        "clients": False,
-                        "broker": False,
-                        "apps": False,
-                        "plugins": False,  # TODO, there is some logic, just enabling plugin does not work, needs also something else
-#                        "with_cjson": False , TODO , adding a dependency is not enough, it needs also be found
+                       "ssl": True,
+                       "clients": False,
+                       "broker": False,
+                       "apps": False,
+                       "cjson": False,
+                       "build_cpp": True,
+                       "websockets": False,
     }
-
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
     _cmake = None
 
-
     @property
     def _source_subfolder(self):
         return "source_subfolder"
 
-    def requirements(self):
-        if self.options.with_tls:
-            self.requires("openssl/1.1.1i")
-        # TODO
-        # if self.options.with_cjson:
-        #     self.requires("cjson/1.7.14")
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
 
     def configure(self):
-        if self.settings.compiler == "Visual Studio" and "MT" in self.settings.compiler.runtime:
-            raise ConanInvalidConfiguration("Visual Studio build for any MT runtime is not supported")
+        if self.options.shared:
+            del self.options.fPIC
+        if not self.options.clients and not self.options.clients:
```

Comment:
I think I am seeing double 😟 

```suggestion
        if not self.options.clients:
```